src/__tests__/index.test.tsx

```diff
@@ -0,0 +1,201 @@
+    render(
+      <div style={{ height: 300 }}>
+        <NetworkGraph data={mockedData} id='network-graph' />
+      </div>,
+    )
+    expect(screen.getByTestId('network-graph')).toBeInTheDocument()
+    expect(screen.getAllByTestId('node').length).toBeGreaterThan(0)
+    expect(screen.getAllByTestId('link').length).toBeGreaterThan(0)
+  })
+
+  it('the graph renders custom nodes and links', () => {
+    render(
+      <div style={{ height: 300 }}>
+        <NetworkGraph
+          data={mockedData}
+          id='network-graph'
+          NodeComponent={CustomNodeComponent}
+          LinkComponent={CustomLinkComponent}
+        />
+      </div>,
+    )
+    expect(screen.getByTestId('network-graph')).toBeInTheDocument()
+
+    // no default nodes and links were rendered
+    expect(screen.queryByTestId('node')).not.toBeInTheDocument()
+    expect(screen.queryByTestId('link')).not.toBeInTheDocument()
+
+    const customNodes = screen.getAllByTestId('custom-node')
+    const customLinks = screen.getAllByTestId('custom-link')
+
+    // custom nodes and links were rendered
+    expect(customNodes.length).toBeGreaterThan(0)
+    expect(customLinks.length).toBeGreaterThan(0)
+
+    const firstNode = screen.getAllByTestId('custom-node')[0]
+
+    // custom nodes renders a label
+    expect(firstNode).toHaveTextContent('Node 1')
+
+    // when a node is clicked, the click handler was triggered
+    fireEvent.click(firstNode)
+    expect(handleNodeClick).toHaveBeenCalled()
+  })
+
+  // it.only('when hovering over a node, the opacity of other nodes changes', () => {
+  //   render(
+  //     <div style={{ height: 300 }}>
+  //       <NetworkGraph
+  //         data={mockedData}
+  //         id='network-graph'
+  //         NodeComponent={CustomNodeComponent}
+  //         LinkComponent={CustomLinkComponent}
+  //         hoverOpacity={0.1}
+  //       />
+  //     </div>,
+  //   )
+
+  //   const firstNode = screen.getAllByTestId('custom-node')[0]
+  //   const secondNode = screen.getAllByTestId('custom-node')[1]
+
+  //   fireEvent.mouseOver(firstNode)
+  //   expect(secondNode).toHaveStyle('opacity: 0.1')
+  // })
+
+  it('renders a detail component when clicking over a node', () => {
+    const CustomNodeComponent = (props: NodeComponentProps<Node>) => {
+      return (
+        <g data-testid='custom-node' onClick={props.selectNode}>
+          <circle r={15} fill={'green'} />
+          <text
+            data-testid='custom-node-text'
+            style={{
+              color: props.isActive ? 'red' : 'green',
+            }}
+            x={20}
+            y={5}
+            onClick={props.onClick}
+          >
+            {props.node.label}
+          </text>
+        </g>
+      )
+    }
+
+    const ObjectDetails = ({ node, unselectNode }: DetailsComponentProps<Node>) => {
+      return (
+        <div role='dialog' className='object-details-wrapper' style={{ maxHeight: 600 - 40 }}>
+          <div className='card'>
```

Comment:
Since this is is just a test, for cleanup purposes, let's remove all `classNames` as they are not important. let's just render

```tsx
<div role='dialog'>
       <h6>{`Details of "${node.label}"`}</h6>
       <button
                title='Close panel'
                data-testid='close-btn'
                onClick={(e) => {
                  e.preventDefault()
                  e.stopPropagation()
                  unselectNode()
                }}
              />
        </div>
      )
```

Reply:
Done